debian/lmi.1
.\" Man page for lmi(1).
.\"
.\" Copyright (C) 2005, 2006, 2007 Vadim Zeitlin
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License version 2 as
.\" published by the Free Software Foundation.
.\"
.\" This program is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License
.\" along with this program; if not, write to the Free Software
.\" Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
.\"
.\" https://savannah.nongnu.org/projects/lmi
.\" email: <chicares@cox.net>
.\" snail: Chicares, 186 Belle Woods Drive, Glastonbury CT 06033, USA
.\"
.\" Please update the date below whenever revising the manpage.
.TH LMI 1 "November  1, 2005"
.\"
.SH NAME
lmi \- Let Me Illustrate... is used to create illustrations for life insurance.
.SH SYNOPSIS
.B lmi_wx
.RI [ options ]
.br
.B lmi_cli
.RI [ options ]
.SH DESCRIPTION
This manual page documents briefly the
.B lmi_wx
and
.B lmi_cli
commands. They provide GUI and command-line interface to the same
functionality: the former is a X11 version while the latter is a command-line
one. We will refer to both of them as
.B lmi
below when the description is applicable to both versions.
.PP
\fBlmi\fP is a program allows you to create illustrations for life insurance.
.SH OPTIONS
These programs follow the usual GNU command line syntax, with long
options starting with two dashes (`-').
A summary of options is included below.
.TP
.B \-a, \-\-accept
Accept license (use \fB-l\fP to see its text first) and don't show its text.
.TP
.B \-c, \-\-cnsfile
Run census file.
.TP
.B \-d, \-\-data_path
Specify the path to data files.
.TP
.B \-i, \-\-illfile
Run illustration.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-l, \-\-license
Show full license text and exit.
.TP
.B \-p, \-\-print_db
Print the product databases.
.TP
.B \-r, \-\-regress
Run the regression test.
.TP
.B \-s, \-\-selftest
Perform self test and exit.
.TP
.B \-t, \-\-test_path
Specify the path to test files.
.SH AUTHOR
lmi was written by Gregory W. Chicares <chicares@cox.net>.
.PP
This manual page was written by Vadim Zeitlin <vadim@wxwindows.org>,
for the Debian project (but may be used by others).


ChangeLog
// Superseded by git: see note at bottom of file.
//
// ChangeLog for "Let me illustrate...". See 'MAINTENANCE' after license.
//
// Copyright (C) 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021 Gregory W. Chicares.
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License version 2 as
// published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software Foundation,
// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
//
// https://savannah.nongnu.org/projects/lmi
// email: <gchicares@sbcglobal.net>
// snail: Chicares, 186 Belle Woods Drive, Glastonbury CT 06033, USA

MAINTENANCE

Please write entries this way:

[1] YYYYMMDDTHHMMZ <yourname@domain.net> [defect-count]
[2] [this line intentionally left blank]
[3]   filename0
[4]   filename1
[5] Terse comments: full English sentences beginning with a capital
    letter and ending with a period. Prefer bare infinitives:
    "Improve ISO 8601 conformance." instead of "Improved...." or
    "The purpose of this change was to improve....". Single-word
    sentences may be appropriate: "Refactor.". Whenever a defect is
    corrected, write "Fix defect introduced YYYYMMDDTHHMMZ." as the
    first sentence, and ponder the questions in 'DefectLog'; write up
    any serious defect in that file.

Limit lines to seventy characters so that they can be quoted and
requoted in email without wrapping.

Explanations:

[1] Write an ISO 8601 timestamp, using UTC because it sorts easily.
Use the actual commit time, so that others can confidently match log
entries against cvs changes. Run 'make check_concinnity' before
committing to cvs: it identifies common problems such as incorrect
copyright statements, and reports the "Number of marked defects" shown
as [defect-count] above.

[3] Exact name of file changed, indented two spaces.

[4] Write each file's name on a separate line, in ASCII sort order:
the same order 'ls' uses, where 'Z' precedes 'a'.

[5] We aren't exactly following GNU standards here, but this one makes
a lot of sense:
  "If you think that a change calls for explanation, you're probably
  right. Please do explain it--but please put the explanation in
  comments in the code, where people will see it whenever they see the
  code."
If, as is often the case, the cvs commit message alone is sufficient,
then just copy it here.

We started following certain conventions only at certain times, e.g.:
  20050710T0414Z: [defect-count]
  20070803T2333Z: "Fix defect introduced YYYYMMDDTHHMMZ."
so don't take earlier entries as exemplary.

//

2005-01-14

Original checkin.

2005-01-15

  actuarial_table.cpp
  crc32.cpp
  crc32_test.cpp
  data_directory.cpp
  ihs_proddata.cpp
Remove carriage returns.

  elapsed_time.cpp
  system_command.cpp
  system_command.hpp
Add missing RCS Id.

2005-01-16

  sample_cgi_census
Expunge this file, which was a disused artifact.

  build.hpp
  expected.cgi.out
  version.hpp
Add copyright and license notices and RCS Ids to these files, no
matter how trivial they seem. Now every single file in this cvs
directory has these things, creating an invariant that can be
confirmed automatically.

2005-01-27 [1092]

  quoted_gpl
  quoted_gpl_html
Add automatically-generated files for embedding the GPL in other
source, as quoted C strings and html respectively.

  GNUmakefile
  autodependency.make
  configuration.make
  msw_common.make
  msw_cygwin.make
  msw_generic.make
  msw_msys.make
  objects.make
  posix_fhs.make
  workhorse.make
Add makefiles.

2005-01-28 [1104]

Changes necessary to compile with gcc-3.4.2 with no diagnostics:

  config.hpp
  platform_dependent.hpp
Make prototypes of nonstandard functions like putenv() available even
in strict standard-conformance mode.

  GNUmakefile
  argv0.hpp
  diagnostics.sed
  miscellany.hpp
  quiet_nan.hpp
  zero.hpp
All non-configuration headers must include "config.hpp" first. Fix
those that didn't. Add a makefile test for this invariant; refine
other such tests slightly, and improve the sed script that parses
makefile output for diagnostics.

  alert.cpp
Remove a defect marker that called for investigation into a problem
whose cause has been tracked to a libstdc++ defect.

  library_demo.cpp
Don't enforce production-grade security for this mere demo.

  financial_test.cpp
Run timing test once only: it took too long with mpatrol.

  security.cpp
Flush message log on passkey validation failure.

  actuarial_table.cpp
  alert.hpp
  basic_values.hpp
  fenv_lmi.hpp
  ihs_acctval.cpp
  ihs_basicval.cpp
  ihs_funddata.hpp
  ihs_ldginvar.cpp
  input_test.cpp
  inputillus.cpp
  inputillus.hpp
  interest_rates.cpp
  ledger_xml_io.cpp
  round_to_test.cpp
  rounding_rules.hpp
  streamable.hpp
  system_command.hpp
  zero.hpp
  zero_test.cpp
Fix other problems diagnosed by gcc-3.4.2, which is stricter than
earlier versions. Use import and export declspec attributes wherever
they make sense: even though it's better not to use them with msw,
they may be useful with gcc-4.x's ELF 'visibility' feature. Remove
superfluous class-name qualifiers from members defined in class
bodies. Factor out duplicative code to manage the floating-point
environment. Other minor changes to improve conformance.

Unit tests do not yet run properly with gcc-3.4.2 and mpatrol, so that
newer compiler version can't yet replace gcc-3.2.3 for production.
However, these changes affect no gcc-3.2.3 test results.

2005-01-31 [1096]

  actuarial_table.cpp
  actuarial_table_test.cpp
  any_member_test.cpp
  cgimain.cpp
  crc32.cpp
  crc32.hpp

lmi.cpp
// Production versus the antediluvian fork.
//
// Copyright (C) 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021 Gregory W. Chicares.
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License version 2 as
// published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software Foundation,
// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
//
// https://savannah.nongnu.org/projects/lmi
// email: <gchicares@sbcglobal.net>
// snail: Chicares, 186 Belle Woods Drive, Glastonbury CT 06033, USA

#include "pchfile.hpp"

#include "lmi.hpp"

bool is_antediluvian_fork()
{
    return false;
}
